main.py

```diff
@@ -1134,16 +1140,25 @@ def handle_link_up(self, interface):
                 link.endpoint_b if link.endpoint_a == interface
                 else link.endpoint_a
             )
-            if other_interface.is_active() is False:
-                self.notify_topology_update()
-                return
-            metadata = {
-                'last_status_change': time.time(),
-                'last_status_is_active': True
-            }
-            link.extend_metadata(metadata)
-            link.activate()
+            if (
+                link.id not in self.link_status_change_cache or
+                not link.is_active()
+            ):
+                status_change_info = self.link_status_change_cache.setdefault(
+                    link.id,
+                    {}
+                )
+                status_change_info['last_status_change'] = time.time()
+                link.activate()
             self.notify_topology_update()
+            if (
+                not other_interface.is_active() or
+                not interface.is_active() or
+                not other_interface.switch.is_active() or
+                not interface.switch.is_active()
+            ):
+                log.info(f"{link} dependencies were not ready.")
```

Comment:
I think this is a great intention, but if you just see a info message `f"{link} dependencies were not ready."` it's too useful why. If you want to keep this and add the value you intend to add, then let's also explain why it wasn't ready and interpolate the conditionals of the ifs here, that'd be handy for us in the future too.

Reply:
Made this a bit more detailed by including the triggering `GenericEntity`.